### PR TITLE
Add KMS support to crypto pseudo-Node

### PR DESCRIPTION
**Pull request by daviddrysdale**
# Checklist

- [x] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [x] I have written tests that cover the code changes.
  - [x] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

**Comment by daviddrysdale** [comment]
Also includes some other fixes for things that got in the way.

File: oak_runtime/Cargo.toml [hunk 1]
```diff
@@ -53,7 +53,10 @@ serde_json = "*"
 sha2 = "^0.9.1"
 tink-core = { version = "^0.1", features = ["json"] }
 tink-aead = "^0.1"
+tink-awskms = "^0.1"
```
Comment (conradgrobler) [inline oak_runtime/Cargo.toml:56]: Is it feasible to make KMS support feature-gated? These KMS support dependencies worry me, and I would prefer to not always include it in the TCB unless someone really wants to use KMS integration.

Comment (daviddrysdale) [reply oak_runtime/Cargo.toml:56]: Good point, will do.

File: oak_runtime/src/node/crypto/tinkwrap.rs [hunk 2]
```diff
@@ -279,6 +326,38 @@ impl TinkWrapper {
...
+
+    fn get_kms_client(
+        &self,
+        key_uri: &str,
+    ) -> Result<std::sync::Arc<dyn tink_core::registry::KmsClient>, tink_core::TinkError> {
+        debug!(
+            "retrieve KMS client for {} using credentials in {:?}",
+            key_uri, self.kms_credentials
+        );
+        #[cfg(feature = "awskms")]
```
Comment (conradgrobler) [inline oak_runtime/src/node/crypto/tinkwrap.rs:338]: Nice. I like how locally contained the feature scoping ended up being.

**Review by conradgrobler — APPROVED** [review]

**Pull request merged**